applications/klighd-cli/src/helpers.ts

```diff
@@ -45,3 +47,22 @@ export function getLanguageId(documentUri: string): string {
 
     return matches[0];
 }
+
+export function getBookmarkViewport(): Viewport | null {
+    const xParam = readSearchParam("bookmarkX");
+    const yParam = readSearchParam("bookmarkY");
+    const zoomParam = readSearchParam("bookmarkZoom");
+
+    if (xParam !== null && yParam !== null && zoomParam !== null) {
+        const x = parseFloat(xParam);
+        const y = parseFloat(yParam);
+        const zoom = parseFloat(zoomParam);
+
+        if (!(isNaN(x) || isNaN(y) || isNaN(zoom))) {
+            return { scroll: { x: x, y: y }, zoom: zoom }
+        }
+    }
+
+    return null
+
+}
```

Comment:
The whole function can't be moved to `klighd-core` as `readSearchParameter` is only available in the CLI.
Should I move just the object construction and NaN check to core or also the floatParsing and the null check?
And where in core would be appropriate?

Reply:
Your are properly right. In the core, it should not have all this parameter checking and "just" expect a number. So nothing would really change, since it mostly consists of massaging parameter...

I would have put it into https://github.com/kieler/klighd-vscode/blob/main/packages/klighd-core/src/base/external-helpers.ts.

Reply:
Ok, moved the check and construction to `external-helper.ts` in core,.
Btw. changed the !isNaN checks to isFinite check as Infinity values should also not be accepted.